src/languages/smtlib2/v2.6/parseSmtlib.mly

```diff
@@ -248,11 +248,16 @@ datatype_dec:
 
 function_dec:
   | OPEN s=SYMBOL OPEN args=sorted_var* CLOSE ret=sort CLOSE
-    { I.(mk term s), args, ret }
+    { I.(mk term s), [], args, ret }
+  | OPEN s=SYMBOL PAR OPEN vars=datatype_symbol+ CLOSE OPEN args=sorted_var* CLOSE ret=sort CLOSE CLOSE
+    { I.(mk term s), vars, args, ret }
```

Comment:
In this new production rule (and for the function definition), you put two CLOSE at the end, which makes the rule not well-parenthesized. From what I can tell, it's probably a mistake that can be solved one of two ways:
- have only one CLOSE parenthesis at the end
- require an OPEN before the PAR

Reply:
I choose the route of adding an `OPEN` token before the `par` as that seemed more coherent with how the rest of the extension works; see branch `psmt2`.